File: mykng/kb-web/src/api/errorLog.ts
import request from './index'
import type { R, PageResult, PageParams } from '@/types'

export interface ErrorLog {
  id: number
  userId?: number
  username?: string
  level: string
  source: string
  message: string
  stackTrace?: string
  url?: string
  ip?: string
  createdAt: string
}

export function reportError(data: {
  level: string
  source: string
  message: string
  stackTrace?: string
  url?: string
}) {
  return request.post<R<void>>('/auth/error-log/report', data)
}

export function getErrorLogList(params: PageParams & {
  level?: string
  source?: string
  startTime?: string
  endTime?: string
}) {
  return request.get<R<PageResult<ErrorLog>>>('/auth/error-log/list', { params })
}

export function getErrorLogDetail(id: number) {
  return request.get<R<ErrorLog>>(`/auth/error-log/${id}`)
}

File: mykng/kb-auth/src/main/java/com/kb/auth/entity/ErrorLog.java
package com.kb.auth.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("sys_error_log")
public class ErrorLog {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String username;
    private String level;
    private String source;
    private String message;
    private String stackTrace;
    private String url;
    private String ip;
    private String userAgent;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
